src/app.js

```diff
@@ -22,26 +23,32 @@ export default class App extends React.Component {
   }
 
   handleSubmit(event) {
-    const card = {}
-
     event.preventDefault()
+    const card = {}
     const createForm = new FormData(event.target)
-    card.question = createForm.get('question')
-    card.answer = createForm.get('answer')
-    this.state.cards.push(card)
-    event.target.reset()
-    window.location.hash = 'card-list'
-    this.setState({
-      path: window.location.hash
-    })
+
+    if (createForm.get('question') === '' || createForm.get('answer') === '') {
+      return null
+    }
+    else {
+      card.question = createForm.get('question')
+      card.answer = createForm.get('answer')
+      this.state.cards.push(card)
+      event.target.reset()
+      window.location.hash = 'card-list'
+      this.setState({ path: window.location.hash })
```

Comment:
You shouldn't need to set the state of the `path` if your `'hashchange'` event listener is working.